### PR TITLE
Update com.android.support libs to androidx. Update test dependencies.

### PR DESCRIPTION
**Pull request by reul**
All the com.android.support libs were replaced by their androidx counterparts and had their versions bumped up.

Deprecated `androidx.test.runner.AndroidJUnit4` is replaced by `androidx.test.ext.junit.runners.AndroidJUnit4` (which now belongs in a separate lib).

Animation scale can no longer be set by `mSetAnimationScalesMethod = windowManagerClazz.getDeclaredMethod("setAnimationScales", float[].class);`, as this method is not present in newer APIs (`android.permission.SET_ANIMATION_SCALE` itself is now a protected permission, restricted to system processes). Instead, `UiAutomation.executeShellCommand(command)` is used for this purpose, with the required permission acquired by UiAutomator at the moment they are needed.

There are still broken tests (wip), but this PR fixes a few of them.

### CODE REVIEW

File: app/src/main/java/org/gnucash/android/util/BackupManager.java
Comment (reul) [inline app/src/main/java/org/gnucash/android/util/BackupManager.java:190]: A few of those can probably be replaced by FLAG_IMMUTABLE, but I haven't tried it yet. Replacing them all causes a crash.

**Comment by djbrown** [comment]
@reul thank you so much for this ❤ I had already moved some of the dependencies and code to androidx, but you got the remaining places.

You can see the remaining failing instrumentation tests under the [actions](https://github.com/GnuCash-Pocket/gnucash-android/actions) section of this repo - couldn't get them to show up under pull request space as they have to be executed via the main branch version of the workflow for security reasons.

Also I had made some other improvements from my personal fork, which I wasn't quite confident to merge into this main repo yet because of the tests still failing (see https://github.com/GnuCash-Pocket/gnucash-android/compare/master...djbrown:gnucash-android:master)

I'd love to hear your opinion on 2808d5b, I couldn't find any solution but to switch to a proper error on the ui.

Looking forward to merging this and if you want I can add you to the organisation or even maintainer.

**Comment by reul** [comment]
Hey, @djbrown! I'd definitely merge those.

What are your thoughts on having a `develop` branch as the default one and keeping `master` for releases only? This way we could keep submitting code to an unstable `develop` and not feeling too bad about it :)

**Comment by Kaligula0** [comment]
> What are your thoughts on having a `develop` branch as the default one and keeping `master` for releases only? This way we could keep submitting code to an unstable `develop` and not feeling too bad about it :)

Looks like a good idea 👍
(BTW, @reul, thanks for your involvement!)

**Comment by reul** [comment]
@djbrown also, any thoughts on [this](https://github.com/reul/gnucash-android/pull/1/files)?

I've noticed navigation was a bit broken (and also a test for TransactionFormFragment) because the deprecated KeyboardView was used instead of a proper input type. I thought it would be easier to just remove all the KeyboardViews and use the software keyboard instead. This way we get the fixed navigation for free.

**Comment by djbrown** [comment]
> What are your thoughts on having a `develop` branch as the default one and keeping `master` for releases only? This way we could keep submitting code to an unstable `develop` and not feeling too bad about it :)

My approach (so far) was to introduce every fix/improvement via pull requests with automated ci (for each change) and merge them into master once they're reviewed. At this point we're just fixing the project, so I'm fine with merging pull requests which still fail some checks.
But I don't know if we get much benefit from a seperate develop branch. We could run different checks/builds for different branches, but I would rahter first try to implement that via matrix workflow and other configuration on one single branch. Also it would be nice if we could rewrite commit history to make the git log of the master more readable, but I think it would create too much problems to be worth. Can you give me some arguments for establishing a develop branch?

Actually I would like to soon merge this pull request into master, since it defenitely improves the overall quality of the project.
Tests aren't still all green, but we're getting there. I'd suggest to create new preferrably isolated/small pull request for the other tests. Of course when improvements are deeply related to another and can't/shouldn't be seperated, then they may very well be combined into a single pull request.

----

> also, any thoughts on [this](https://github.com/reul/gnucash-android/pull/1/files)?
>
> I've noticed navigation was a bit broken (and also a test for TransactionFormFragment) because the deprecated KeyboardView was used instead of a proper input type. I thought it would be easier to just remove all the KeyboardViews and use the software keyboard instead. This way we get the fixed navigation for free.
>
> I'm PR-ing my own branch to keep that concern separated from this PR (which is getting big).

https://github.com/reul/gnucash-android/pull/1 could/should just as well be a separate pull request to the main repo in that regard. So far it seems good to me and might be the best next step to improve.

----

> I'd definitely merge those.

Thanks @reul for the review! I'll create a pull request, except for the toast testing bit, since it only changed the test code but the main code needs to be updated from toast to input error:
https://github.com/GnuCash-Pocket/gnucash-android/blob/3cc49e23b42a90cba2fad94a78c82a53b1d8dc93/app/src/main/java/org/gnucash/android/ui/transaction/TransactionFormFragment.java#L971

**Comment by reul** [comment]
> My approach (so far) was to introduce every fix/improvement via pull requests with automated ci (for each change) and merge them into master once they're reviewed. At this point we're just fixing the project, so I'm fine with merging pull requests which still fail some checks.
> But I don't know if we get much benefit from a seperate develop branch. We could run different checks/builds for different branches, but I would rahter first try to implement that via matrix workflow and other configuration on one single branch. Also it would be nice if we could rewrite commit history to make the git log of the master more readable, but I think it would create too much problems to be worth. Can you give me some arguments for establishing a develop branch?

That definitely works for me. We can figure out a branching strategy after the first release, when the project is fixed.


> Actually I would like to soon merge this pull request into master, since it defenitely improves the overall quality of the project.
> Tests aren't still all green, but we're getting there. I'd suggest to create new preferrably isolated/small pull request for the other tests. Of course when improvements are deeply related to another and can't/shouldn't be seperated, then they may very well be combined into a single pull request.

Yeah, definitely. My PR from the other repo is dependent on this one, but I can definitely send a separate PR once this is merged.

I've kinda figured out what's wrong with ExportTransactionsTest as well. As soon as I can fix it, I will send yet another PR.

File: app/build.gradle
Comment (djbrown) [inline app/build.gradle:175]: `androidSupportVersion` seems unused now, can we remove it?

**Review by djbrown — APPROVED** [review]
LGTM, just some probably unused variable in app/buildgradle.

**Pull request merged**
